lib/collection/src/collection_manager/optimizers/segment_optimizer.rs

```diff
@@ -371,6 +372,15 @@ pub trait SegmentOptimizer {
 
         let mut optimized_segment: Segment = segment_builder.build(permit, stopped)?;
 
+        // Make the IdTracker immutable if built segment is not appendable.
+        if !optimized_segment.appendable_flag {
+            let immutable = optimized_segment
+                .id_tracker
+                .borrow()
+                .make_immutable(&optimized_segment.current_path)?;
+            optimized_segment.id_tracker = Arc::new(AtomicRefCell::new(immutable));
+        }
```

Comment:
This should rather happen inside `segment_builder.build`

Reply:
`segment_builder` should persist all files for immutable id_tracker, and then load segment with proper id_tracker based on configuration. 

In other words, content of the segment should always be defined by it's config, and we should not patch is afterwards